google-codelab-survey/google_codelab_survey.js

```diff
@@ -33,33 +66,118 @@ class CodelabSurvey extends HTMLElement {
      * The name of the survey
      * @private {string}
      */
-    this.surveyName_ = this.getAttribute('survey-id') || '';
+    this.surveyName_ = this.getAttribute(SURVEY_ID_ATTR) || DEFAULT_SURVEY_NAME;
+
+    /** @private {!HTML5LocalStorage} */
+    this.storage_ = new HTML5LocalStorage();
+
+    /** @private {string} */
+    this.storageKey_ = STORAGE_KEY_PREFIX + this.surveyName_;
+
+    /** @private {!Object<string, !Object>} */
+    this.storedData_ = {};
+
+    /** @private {!EventHandler} */
+    this.eventHandler_ = new EventHandler();
   }
 
   /**
    * @export
    * @override
    */
   connectedCallback() {
+    this.checkStoredData_();
     this.updateDom_();
+    this.bindEvents_();
+  }
+
+  bindEvents_() {
+    const surveyQuestions = document.querySelectorAll(
+      Selectors.OPTIONS_WRAPPER);
+    surveyQuestions.forEach(el => {
```

Comment:
What about a single event listener on the parent?

Reply:
Done!